store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -5773,7 +5773,8 @@ private Message addMessage(OperationContext octxt, ParsedMessage pm, int folderI
         final MailboxLock l = lockFactory.writeLock();
         try {
             ZimbraLog.mailbox.info("addMessage tries to get lock");
-            l.lock();
+			// "addMessageInternal" method calls MailboxTRansaction and inside of it, is called lock(), if this line is not commented we should have 2 locks and that is incorrect
+            //l.lock(); 
```

Comment:
@Raffaell0 I think it is ok to call `l.lock()` multiple times. If the current thread holds the lock than calling `.lock` should be a noop.

Reply:
multiple instances are created if you call lock()